### ISSUE TITLE
Misleading behavior when ~/.kube/config uses a non-existing context

### ISSUE THREAD

**Issue by mwielgus** [GoogleCloudPlatform/kubernetes#10616]
There are 2 problems here:

1. `kubectl config use-context foobar`. Should throw an error or at least print a warning if foobar is not defined.
2. `kubectl get pods` with non-existing current-context . There is no server to talk to but the error is:
```
error: couldn't read version from server: the server could not find the requested resource
```

**Comment by bprashanth** [comment]
This is a really confusing error

**Comment by jlowdermilk** [comment]
1. Was fixed by @mikedanese in #10510

2. This is less of a problem as the only way to set an invalid current-context is to edit kubeconfig out of band. If you do manage that, it will default to using local server
```
error: couldn't read version from server: Get http://localhost:8080/api: dial tcp 127.0.0.1:8080: connection refused
```

@mwielgus the error you saw likely resulted from having an older version of local kube-apiserver running when you tried `kubectl get` with nonexistent current-context. I think we can call this fixed.
